sound/soc/sof/topology.c

```diff
@@ -1577,6 +1583,12 @@ static int sof_effect_iir_load(struct snd_soc_component *scomp, int index,
 		le32_to_cpu(private->size) +
 		iir_data_size;
 
+	if (ipc_size > SOF_IPC_MSG_MAX_SIZE) {
+		iir_data_size = 0;
+		ipc_size = sizeof(struct sof_ipc_comp_eq_iir) +
+			le32_to_cpu(private->size);
+	}
+
```

Comment:
this looks very odd - and I don't get why we have eq-specific logic in the topology in the first place.
@ranj063 @singalsu why are we doing all this? This should be common for all effects

Reply:
@plbossart this is probably also partly written by me. however we had special ipc message struct for fir and iir. If we decide that all effects have common part that is the same and other data is going in binary blob, that's fine. I guess the idea before was that we have other specialized array data in different effects, so we need differentiation.

Reply:
The EQs have very similar configuration IPC so they could both use "struct sof_ipc_comp_effect". I'd assume it should work for other effects too. Should I do a separate PR for it or can we do the cleanup in this PR?

Reply:
let's unify the effect structs in another pr

Reply:
@juimonen Liam's PR already unifies the effect's type into a new type called process. 
Also, depending on which one gets merged first, this PR might need some adjustments.